### PR TITLE
Add force_pkce option

### PR DESCRIPTION
**Pull request by mattmanning**
The force_pkce option is disabled by default. When enabled, it requires non-confidential clients to use PKCE when requesting an access_token using an authorization code.

### Summary

PKCE should be used for all non-confidential clients (those who can't protect their client_secret). However, since the OAuth 2.0 spec doesn't require it, it should be disabled by default and only used if desired by the server. https://www.rfc-editor.org/rfc/rfc7636#section-5

### Other Information

This implementation is slightly different than what was discussed in https://github.com/doorkeeper-gem/doorkeeper/issues/1654. A more stringent option that also requires PKCE for confidential clients could later be added to a bundled OAuth 2.1 option.

### CODE REVIEW

**Review by anolson — COMMENTED** [review]
👍

**Comment by mattmanning** [comment]
@nbulaj Any feedback on this? We've now implemented it as monkeypatches in my company's app, but it would be cool to get it into the gem.

I'm happy to make changes to the design or the implementation if you have any issues with the current code.

Thanks!

File: doorkeeper.gemspec [hunk 1]
```diff
@@ -52,6 +52,7 @@ Gem::Specification.new do |gem|
   gem.add_development_dependency "factory_bot", "~> 6.0"
   gem.add_development_dependency "generator_spec", "~> 0.10.0"
   gem.add_development_dependency "grape"
+  gem.add_development_dependency "pry-byebug"
```
Comment (nbulaj) [inline doorkeeper.gemspec:55]: I don't find pry / byebug useful, can we avoid adding it there please?

**Review by nbulaj — COMMENTED** [review]
Just a minor comment, otherwise LGTM :+1:

**Comment by mattmanning** [comment]
@nbulaj Thank you for looking!

I have removed pry-byebug. Please let me know if you would like any other changes.

**Review by nbulaj — APPROVED** [review]
LGTM :bowing_man:

**Pull request merged**